### PR TITLE
Views

### PR DESCRIPTION
**Pull request by rbd3**
### Change made:
👉 Add index for user
👉 Add show for user
👉 Add index for post
👉 Add show for post

### CODE REVIEW

File: app/views/users/_header.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div class="container-header">
+  <img src="<%= user.photo %>" class="user-photo">
+  <div class="user-card-info">
+    <div class="flex-col">
+      <span class="header-title"><%= user.name %></span>
+      <span class="user-totalpost">Number of posts: <%= user.posts_counter %></span>
+    </div>
+  </div>
+</div>
```
Comment (julie-ify) [inline app/views/users/_header.html.erb]: - [ ] Kindly ensure your page layout matches [the design](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_root_users.png). Styling is not a requirement for this project, but the page layouts should be maintained.

- [ ] Every user should also be a link to their respective user show page.

  image

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,2 +1,27 @@
...
+</div>
+
+<a 
+  href="<%= user_posts_path(user_id: @user.id) %>" 
+  class="user-card" 
+> 
+  <button class="see-all-posts">
+  See All Posts
+  </button>  
+</a>
```
Comment (julie-ify) [inline app/views/users/show.html.erb]: - [ ] Kindly ensure the user show page uses [the layout design](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_page.png).

  image

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,2 +1,14 @@
-<h1>Hi</h1>
-
+<%= render partial: "users/header", locals: { user: @user } %>
+<% @posts.each do |post| %>
+  <%= render partial: "post", locals: { post: post } %>
+  
+  <div class="recent-comments">
+  <% if post.recent_comments.any? %>
+    <% post.recent_comments.each do |comment| %>
+      <p><%= comment.user.name %>: <%= comment.text %></p>
```
Comment (julie-ify) [inline app/views/posts/index.html.erb]: - [ ] There is an error when I visit the post index page. Kindly fix it. This should be `author`, not `user`, according to your model.

- [ ] Each post on the index page should also be a link to the post's show page.

- [ ] Also ensure the design follows [the layout](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png)

  image

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -1,2 +1,24 @@
...
+      <p>Likes: <%= @post.likes.count %></p>
+    </div>
+
+    <p><%= @post.text %></p>
+  </div>
+
+  <div class = "all-comments">
+  <% if @post.comments.any? %>
+    <% @post.comments.each do |comment| %>
+      <p><%= comment.user.name %>:
```
Comment (julie-ify) [inline app/views/posts/show.html.erb]: - [ ] I also get errors when I visit the post show page. Kindly fix it. This should be `author`, not `user`, according to your model.

- [ ] Also ensure the page wireframe conforms to [the design](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_single_post.png).

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @rbd3,

Good job so far!

## Highlights!
- Good partial use ✔️ 
- Good model methods use in views ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @rbd3    , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  . 
Congratulations! 🎉  Keep up the Awesome work 🥂  Keep Rocking 🥇 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] Professional README file 
- [x] App running as intended. 💯 🥇 
- [x] Views Correctly implemented 




 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**